pkg/operator/controller/ingress/monitoring.go

```diff
@@ -34,16 +34,18 @@ func (r *reconciler) ensureServiceMonitor(ic *operatorv1.IngressController, svc
 			return false, nil, fmt.Errorf("failed to create servicemonitor %s/%s: %v", desired.GetNamespace(), desired.GetName(), err)
 		} else if created {
 			log.Info("created servicemonitor", "namespace", desired.GetNamespace(), "name", desired.GetName())
+			return r.currentServiceMonitor(ic)
 		}
```

Comment:
If `err == nil && !created`, then control continues on to `return true, current, nil` at the end of the method, and `current` is still nil.  Simplest fix would be to move the return statement on line 37 down a line (outside the `else if`).  I wouldn't object if you decided to delete `createServiceMonitor` and inline its logic here.

Reply:
Fixed, went with inline logic to save some lines.

Reply:
Just to be clear, previously `createServiceMonitor` was only capable of returning `false, err` or `true, nil`, so the case of `err == nil && !created` would not have been possible. However, that is in no way obvious, so using inline logic is definitely the way to go. Thanks!